### DISCUSSION TITLE
Event when multiple cells are updated?

### DISCUSSION THREAD

**Discussion by chuckurbis** [nick-keller/react-datasheet-grid#94] [Q&A] [answered]
Is there a way of knowing when multiple cells are updated? 

Use case: The data we are displaying in our table is stored remotely and we are trying to minimize the number of times the data is saved. Currently, we set the row data to a state variable, then listen to onBlur events for when to save the data to our server. But blur events don't happen when cells are added, etc. 

I do see all the data change in the onChange event, but other than comparing the original to the updated change, I have no way of knowing if one cell was changed, or many. My worry is, on a large table, doing a deep check of the array will be very slow. 

Not sure the best solution in this case. Perhaps a delta in the onChange event, or even just a value indicating the type (input, paste, duplication, new row, etc)?

Are there any methods or callbacks I am not considering to achieve this?

**Comment by nick-keller** [comment]
Hey @chuckurbis ,
I would suggest doing some kind of diff yourself. You can see an example here: https://github.com/Equify/react-datasheet-grid/discussions/81#discussioncomment-1674351
You should of course do some tests on performance.

Now to elaborate, there are many ways as you said to update cells, copy-pasting, editing, deleting cells, deleting rows, expanding selection, inserting rows... Another very important thing to consider is that the underlying data does not have to match the concept of cells, you can have multiple cells impacting the same key of your object, or have a cell impact multiple keys.

But doing this with rows sounds feasible as there is a 1 to 1 relationship between the data and the rows displayed.

Maybe we could add a second parameter to the `onChange` callback that would tell you if rows were added, deleted, or updated. How would that sound?

**Reply by chuckurbis** [reply]
Yeah I think that would be a useful feature. Right now, I'm just doing a debounced auto save on it instead, but it's a little nondeterministic which I hate. Thanks for the quick reply and awesome library.

**Comment by nick-keller** [comment, marked as answer]
This is now available in 3.4.0, please read the doc for [`onChange`](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onchange)
